# Etape 1 : Exploration des données + DataViz’

Attendue de cette étape :

- Votre première tâche consistera à définir le contexte et le périmètre du projet : j’attends que vous preniez vraiment le temps de bien comprendre le projet et de vous renseigner au mieux sur les notions que celui-ci va introduire.

- Il faudra ensuite prendre en main et découvrir votre jeu de données et faire une analyse presque exhaustive de celui-ci afin de mettre en lumière la structure, les difficultés et eventuels biais du dataset.

J’attendrai également au moins 5 représentations graphiques construites à partir de votre jeu de données, visuelles et surtout pertinentes. Pour chacune d’elle j’attendrai :
- Un commentaire précis, qui analyse la figure et apporte un avis “métier”.
- Une validation du constat par des manipulations de données, ou un test statistique.

## Contexte et périmètre du projet

Le projet "Compagnon_Immo" a pour but de développer une solution innovante permettant aux acheteurs et investisseurs immobiliers d’explorer, comparer et estimer les prix des logements sur différents territoires.

Dans un premier temps nous nous focalisons sur l'analyse des données immobilières du Haut-Rhin (Département 68). Ces données comportent des informations sur des biens en locations et en ventes dans le département entre 2019 et 2023.

Nous avons à notre disposition une cinquentaine de variables pour prédire notre cible, le prix du bien.
Ces variables décrives les propriétés des biens comme leur surface, le nombre de pièces, d'étages etc ... mais aussi bien leur années de construction et leur emplacement géographique. Tout ces éléments impactent le prix du bien, tachons maintenant d'identifier les variables pertinantes pour notre analyse.

### Description des jeux de données

Voici un descriptif des variables disponibles dans nos jeux de données.

Jeu de données des locations dans le Haut-Rhin :

In [6]:
import pandas as pd

df_locations = pd.read_csv('data/ech_annonces_locations_68.csv', sep=';', index_col='idannonce')
df_ventes = pd.read_csv('data/ech_annonces_ventes_68.csv', sep=';', index_col='idannonce')

In [4]:
df_locations.columns.to_list()

['type_annonceur',
 'typedebien',
 'typedetransaction',
 'etage',
 'surface',
 'surface_terrain',
 'nb_pieces',
 'prix_bien',
 'prix_maison',
 'prix_terrain',
 'mensualiteFinance',
 'balcon',
 'eau',
 'bain',
 'dpeL',
 'dpeC',
 'mapCoordonneesLatitude',
 'mapCoordonneesLongitude',
 'annonce_exclusive',
 'nb_etages',
 'parking',
 'places_parking',
 'cave',
 'exposition',
 'ges_class',
 'annee_construction',
 'nb_toilettes',
 'nb_terraces',
 'videophone',
 'porte_digicode',
 'surface_balcon',
 'ascenseur',
 'nb_logements_copro',
 'charges_copro',
 'chauffage_energie',
 'chauffage_systeme',
 'chauffage_mode',
 'categorie_annonceur',
 'logement_neuf',
 'duree_int',
 'typedebien_lite',
 'date',
 'INSEE_COM',
 'IRIS',
 'CODE_IRIS',
 'TYP_IRIS_x',
 'TYP_IRIS_y',
 'GRD_QUART',
 'UU2010',
 'REG',
 'DEP']

In [11]:
df_ventes.columns.to_list()

['type_annonceur',
 'typedebien',
 'typedetransaction',
 'etage',
 'surface',
 'surface_terrain',
 'nb_pieces',
 'prix_bien',
 'prix_maison',
 'prix_terrain',
 'mensualiteFinance',
 'balcon',
 'eau',
 'bain',
 'dpeL',
 'dpeC',
 'mapCoordonneesLatitude',
 'mapCoordonneesLongitude',
 'annonce_exclusive',
 'nb_etages',
 'parking',
 'places_parking',
 'cave',
 'exposition',
 'ges_class',
 'annee_construction',
 'nb_toilettes',
 'nb_terraces',
 'videophone',
 'porte_digicode',
 'surface_balcon',
 'ascenseur',
 'nb_logements_copro',
 'charges_copro',
 'chauffage_energie',
 'chauffage_systeme',
 'chauffage_mode',
 'categorie_annonceur',
 'logement_neuf',
 'duree_int',
 'typedebien_lite',
 'date',
 'INSEE_COM',
 'IRIS',
 'CODE_IRIS',
 'TYP_IRIS_x',
 'TYP_IRIS_y',
 'GRD_QUART',
 'UU2010',
 'REG',
 'DEP',
 'loyer_m2_median_n6',
 'nb_log_n6',
 'taux_rendement_n6',
 'loyer_m2_median_n7',
 'nb_log_n7',
 'taux_rendement_n7',
 'prix_m2_vente']

In [5]:
df_locations.describe()

,etage,surface,surface_terrain,nb_pieces,prix_bien,prix_maison,prix_terrain,mensualiteFinance,balcon,eau,...,nb_logements_copro,charges_copro,duree_int,INSEE_COM,IRIS,CODE_IRIS,GRD_QUART,UU2010,REG,DEP
count,11702.000000,11702.000000,507.000000,11702.000000,11702.000000,0.0,0.0,11702.0,11702.000000,11702.000000,...,9.000000,0.0,11448.000000,11702.000000,11702.000000,1.170200e+04,1.170200e+04,11702.000000,11702.000000,11702.000000
mean,1.840711,65.150060,564.100118,2.820629,714.543924,NaN,NaN,0.0,0.186977,0.286789,...,18.666667,NaN,-207.466632,68191.909844,244.231499,6.819193e+08,6.819192e+06,68324.462400,44.018886,67.999316
std,24.546138,30.798055,776.823114,1.219675,286.924335,NaN,NaN,0.0,0.459904,0.465137,...,42.272923,NaN,422.747918,814.069476,385.032725,8.140732e+06,8.140730e+04,3166.426849,1.763585,0.807832
min,-2.000000,7.000000,1.710000,1.000000,1.000000,NaN,NaN,0.0,0.000000,0.000000,...,0.000000,NaN,-1125.000000,6088.000000,0.000000,6.088010e+07,6.088010e+05,758.000000,11.000000,6.000000
25%,0.000000,45.000000,200.000000,2.000000,531.000000,NaN,NaN,0.0,0.000000,0.000000,...,3.000000,NaN,-757.000000,68093.000000,0.000000,6.809300e+08,6.809300e+06,68402.000000,44.000000,68.000000
50%,1.000000,62.000000,455.000000,3.000000,660.000000,NaN,NaN,0.0,0.000000,0.000000,...,5.000000,NaN,15.000000,68224.000000,103.000000,6.822402e+08,6.822401e+06,68501.000000,44.000000,68.000000
75%,2.000000,79.000000,700.000000,3.000000,820.000000,NaN,NaN,0.0,0.000000,1.000000,...,8.000000,NaN,41.000000,68271.000000,202.000000,6.827101e+08,6.827101e+06,68701.000000,44.000000,68.000000
max,999.000000,450.000000,10198.000000,13.000000,3287.000000,NaN,NaN,0.0,12.000000,3.000000,...,131.000000,NaN,504.000000,86214.000000,6220.000000,8.621401e+08,8.621401e+06,86601.000000,93.000000,86.000000


In [8]:
df_ventes.describe()

,etage,surface,surface_terrain,nb_pieces,prix_bien,prix_maison,prix_terrain,mensualiteFinance,balcon,eau,...,UU2010,REG,DEP,loyer_m2_median_n6,nb_log_n6,taux_rendement_n6,loyer_m2_median_n7,nb_log_n7,taux_rendement_n7,prix_m2_vente
count,27737.00000,27737.000000,11444.000000,27737.000000,2.773700e+04,437.000000,437.000000,27737.000000,27737.000000,27737.000000,...,27737.000000,27737.0,27737.0,17485.000000,17485.000000,17485.000000,19650.000000,19650.000000,19650.000000,27737.000000
mean,1.39204,108.343188,1017.414037,4.526301,2.773427e+05,213134.462243,89757.581236,13.232433,0.167610,0.283809,...,68370.913834,44.0,68.0,10.779532,17.011210,0.056380,10.639302,151.035929,0.056315,2624.974439
std,26.84627,60.845498,6694.219735,2.029696,1.861352e+05,71437.516537,47633.302968,139.079554,0.415649,0.541601,...,269.815128,0.0,0.0,2.066664,29.391785,0.083427,1.946427,252.454136,0.081040,1195.389487
min,-1.00000,2.000000,1.000000,1.000000,1.076000e+03,86000.000000,1.000000,0.000000,0.000000,0.000000,...,68000.000000,44.0,68.0,3.080000,1.000000,0.003000,4.780000,1.000000,0.003000,15.590000
25%,0.00000,71.000000,378.000000,3.000000,1.590000e+05,156500.000000,62000.000000,0.000000,0.000000,0.000000,...,68112.000000,44.0,68.0,9.440000,2.000000,0.038000,9.320000,3.000000,0.038000,1870.000000
50%,0.00000,95.000000,589.000000,4.000000,2.420000e+05,199620.000000,87000.000000,0.000000,0.000000,0.000000,...,68403.000000,44.0,68.0,10.530000,6.000000,0.048000,10.380000,18.000000,0.048000,2580.650000
75%,1.00000,130.000000,875.000000,5.000000,3.470000e+05,250010.000000,115800.000000,0.000000,0.000000,1.000000,...,68701.000000,44.0,68.0,11.950000,19.000000,0.066000,11.810000,190.000000,0.066000,3268.480000
max,999.00000,980.000000,431600.000000,43.000000,3.380000e+06,514999.000000,320000.000000,7715.000000,8.000000,10.000000,...,68701.000000,44.0,68.0,23.730000,212.000000,10.496000,23.730000,960.000000,10.773000,79500.000000


La variable catégorielle 'type_annonceur' contient les modalités pr et pa pour PRO et PARTICULIER.

La variable catégorielle 'typedebien' contient les modalités m, a, l, mn et an pour Maison, Appartement, Lot, (Maison ?) et (Appartement ?).

La variable catégorielle 'typedetransaction' contient les modalités l, lt, v, vp et pi pour Location, (lt : type de location ?), Vente, (vp : type de vente?) et (pi ?).

La variable numérique 'etage' correspond à l'étage où se situe le bien. Pour les locations : les valeurs vont de -2 à 999 mais la médiane est de 1 et la moyenne de 1.8. Pour les ventes : les valeurs vont de -1 à 999 mais la médiane est de 0 et la moyenne de 1.3.

La variable numérique 'surface' correspond à la superficie en m² du bien. Pour les locations : les valeurs vont de 7 à 450 avec une médiane de 62 et une moyenne de 65.1. Pour les ventes : les valeurs vont de 2 à 980 avec une médiane de 95 et une moyenne de 108.3.

La variable numérique 'surface_terrain' correspond à la superficie en m² du terrain associé au bien. Pour les locations : les valeurs vont de 1.7 à 10198 avec une médiane de 455 et une moyenne de 564.1. Pour les ventes : les valeurs vont de 1 à 431600 avec une médiane de 589 et une moyenne de 1017.4.

La variable numérique 'nb_pieces' correspond aux nombres de pièces dans le bien. Pour les locations : les valeurs vont de 1 à 13 avec une médiane de 3 et une moyenne de 2.8. Pour les ventes : les valeurs vont de 1 à 43 avec une médiane de 4 et une moyenne de 4.5.

TODO :
- 'prix_bien',
- 'prix_maison',
- 'prix_terrain',
- 'mensualiteFinance',
- 'balcon',
- 'eau',
- 'bain',
- 'dpeL',
- 'dpeC',
- 'mapCoordonneesLatitude',
- 'mapCoordonneesLongitude',
- 'annonce_exclusive',
- 'nb_etages',
- 'parking',
- 'places_parking',
- 'cave',
- 'exposition',
- 'ges_class',
- 'annee_construction',
- 'nb_toilettes',
- 'nb_terraces',
- 'videophone',
- 'porte_digicode',
- 'surface_balcon',
- 'ascenseur',
- 'nb_logements_copro',
- 'charges_copro',
- 'chauffage_energie',
- 'chauffage_systeme',
- 'chauffage_mode',
- 'categorie_annonceur',
- 'logement_neuf',
- 'duree_int',
- 'typedebien_lite',
- 'date',
- 'INSEE_COM',
- 'IRIS',
- 'CODE_IRIS',
- 'TYP_IRIS_x',
- 'TYP_IRIS_y',
- 'GRD_QUART',
- 'UU2010',
- 'REG',
- 'DEP'
- 'loyer_m2_median_n6',
- 'nb_log_n6',
- 'taux_rendement_n6',
- 'loyer_m2_median_n7',
- 'nb_log_n7',
- 'taux_rendement_n7',
- 'prix_m2_vente'

### Variables selectionnées pour la première analyse

Après une prise de connaissance des contenus des fichiers csv, de la liste des colonnes, de leurs valeurs, distribution et répartition - nous avons fait le choix de nous focaliser sur les variables suivantes :
- localisation
- surface
- etages
- type de bien
- nb de pièces

## DataViz'

#### Données de locations

#### Données de ventes